tests/test_ir.py

```diff
@@ -162,6 +162,39 @@ def test_region_clone_into_circular_blocks():
     assert region.is_structurally_equivalent(region2)
 
 
+def test_non_empty_block_with_parent_region_requires_terminator():
+    """
+    Tests that an empty block belonging to a single-block region with parent
```

Comment:
```suggestion
    Tests that an empty block belonging to a multi-block region with parent
```
? Not sure if this what you meant